apps/chat/src/store/prompts/prompts.selectors.ts

```diff
@@ -211,6 +215,20 @@ export const hasExternalParent = createSelector(
   },
 );
 
+export const selectExternalEntities = createSelector(
+  [selectFolders, selectPrompts, (state) => state],
+  (folders, prompts, state) => {
+    return {
+      externalPrompts: prompts.filter((prompt) =>
+        isEntityOrParentsExternal(state, prompt, FeatureType.Prompt),
+      ),
+      externalFolders: folders.filter((folder) =>
+        isEntityOrParentsExternal(state, folder, FeatureType.Prompt),
+      ),
+    };
+  },
+);
+
```

Comment:
Why do we do all this check on UI?
Let's memorise it in selector `selectDoesAnyMyItemExists` which returns boolean instead
```suggestion
export const selectDoesAnyMyItemExist = createSelector(
  [selectFolders, selectPrompts],
  (folders, prompts) => {
    const conversationRootId = getPromptRootId();
    return (
      prompts.some((prompt) => prompt.id.startsWith(conversationRootId)) ||
      folders.some((folder) => folder.id.startsWith(conversationRootId))
    );
  },
);
```